### ISSUE TITLE
unable to run

### ISSUE THREAD

**Issue by lyris85** [smourier/VCamSample#1]
![Screenshot 2024-04-14 132610](https://github.com/smourier/VCamSample/assets/17769112/6a23e65e-7faf-43ef-afdc-c78768b6ec32)

I tried on 2 different computer and it does the same problem. I registred the .dll with regsvr32 VCamSampleSource.dll as administrator

**Comment by smourier** [comment]
I just tried on a Windows 11 Sandbox (https://learn.microsoft.com/en-us/windows/security/application-security/application-isolation/windows-sandbox/windows-sandbox-overview) and on a brand new Windows 11 VM from the ISO here https://www.microsoft.com/en-us/software-download/windows11 and it works fine.

So  I guess it's related to your security settings or more generally how your machine is configured, but I'm unsure what's exactly the problem. There are multiple processes that needs access:

* VCamSample.exe
* SvcHost.exe which runs two processes: the frame server and the frame server monitor like we can see here:

![image](https://github.com/smourier/VCamSample/assets/5328574/0533049e-c1e7-441f-a965-def594d8cdf8)

Access denied at that point is usually because these two services have a problem to run, or they can't access the registry of the binaries where the VCamSampleSource was registered.

Oh and ... maybe it's due the fact that you downloaded the binaries from the internet, not compiled them by yourself. So you must remove the Mark of the Web (https://en.wikipedia.org/wiki/Mark_of_the_Web) click on "Unblock" on the .zip file you downloaded:

![image](https://github.com/smourier/VCamSample/assets/5328574/14a57db7-ad4b-472f-8549-4dbde37c4a76)

**Comment by lyris85** [comment]
I tried again using your recommandation in window sandox, same result: access denied.

Here is the config I used to build the project:
![Screenshot 2024-04-15 073429](https://github.com/smourier/VCamSample/assets/17769112/1d5139a4-cd05-46b4-b17c-b396a26e36a2)
I tried all config ( x86 debug and release, x64 debug and release)

Here is the output build:
![Screenshot 2024-04-15 074220](https://github.com/smourier/VCamSample/assets/17769112/8e029aae-d5de-4ddb-b78c-32d47b0a250b)

I have Windows 11 Pro version 23H2

**Comment by smourier** [comment]
Have you registered this new VCamSampleSource.dll output files?

**Comment by lyris85** [comment]
Yes I did it lik
![Untitled](https://github.com/smourier/VCamSample/assets/17769112/83081a48-14de-4945-a74e-9cbc478d5747)
e this:

**Comment by smourier** [comment]
It works fine (downloaded or rebuilt) in a brand new VM with these specs:

![image](https://github.com/smourier/VCamSample/assets/5328574/d2868053-4799-47e0-9248-937b04551b98)

Security issues can arise from many causes and Windows 11 Virtual Cam involves at least 3 different processes that all need registry (read) access. Difficult to say w/o being able to reproduce.

You can try to use procmon https://learn.microsoft.com/en-us/sysinternals/downloads/procmon to check access denieds (on files and/or registry keys), and/or build both binaries in Debug and check the traces the sample outputs, using what's described here https://github.com/smourier/VCamSample?tab=readme-ov-file#tracing

**Comment by smourier** [comment]
Ok I think the problem is the **Frame Server**  & **Frame Server Monitor** services (running as `svchost.exe` with special command lines) must load `VCamSampleSource.dll` in-process and  they're running under different credentials, so they just get an access denied when they load this dll from where you compile, and this access denied is then propagated to VCamSample which it calls Start.

![image](https://github.com/smourier/VCamSample/assets/5328574/7bc18f95-8ebf-41e9-959a-5f676ee3fc0b)

I guess you compile in a directory under your compiling user's root, for example something like `C:\Users\<your login>\source\repos\VCamSample\x64\Debug` or somewhere restricted in some way.

It works with the version I built because I suppose you download it (and register it there) under a directory that's accessible by the two Frame Server services.

So the solution is just to copy `VCamSampleSource.dll` once built somewhere where everyone has access and register it from here (or build and register where everyone has access)

**Comment by My-zhai** [comment]
按照你给出的方法，我成构建并运行，由于是第一次使用c++ 还有好多不明白的地方，
在win11内 相机使用的是GPU

![屏幕截图 2024-04-19 141401](https://github.com/smourier/VCamSample/assets/160556190/30fcfe7e-9eac-44c2-acde-b9800dec521c)

但是obs使用的是CPU，如何让obs也使用GPU，

![Screenshot 2024-04-19 14-17-13](https://github.com/smourier/VCamSample/assets/160556190/c5bbe706-bf40-4128-bc13-5c2827755ef3)
控制颜色的代码_renderTarget->Clear(D2D1::ColorF(0.2f, 0.3f, 0.5f, 0.5f)); 
从两张图片可以明显的看出，透明度在相机内生效，但是在obs 内却无效的

**Comment by smourier** [comment]
Hi, I don't speek chinese (?), and I don't think this is related to the issue. Please delete it from here and repost it in english as another issue (or I will delete it myself)
